src/webnn/native/dml/GraphDML.cpp

```diff
@@ -1719,7 +1767,170 @@ namespace webnn::native { namespace dml {
+                    return DAWN_INTERNAL_ERROR("Failed to create DML tensor description.");
+                }
+                inputTensorDesc = {DML_TENSOR_TYPE_BUFFER, &inputDmlTensorDesc->bufferDesc};
+            }
+
+            DML_TENSOR_DESC filterTensorDesc = filterEdge->outputTensorDESC;
+            if (options->filterLayout != wnn::ConvTranspose2dFilterOperandLayout::Iohw) {
+                newFilterDims = transposeFilterDimensionsAsIohw(options->filterLayout, filterDims);
+                newFilterStrides = transposeFilterStridesAsIohw(options->filterLayout, filterDims);
+
+                std::shared_ptr<DmlTensorDesc> filterDmlTensorDesc(new DmlTensorDesc);
+                if (!CreateDmlTensorDesc(mDmlTensorsDesc, filterDmlTensorDesc,
+                                         &filterEdge->outputTensorDESC, newFilterDims,
+                                         newFilterStrides)) {
+                    return DAWN_INTERNAL_ERROR("Failed to create DML tensor description.");
+                }
+                filterTensorDesc = {DML_TENSOR_TYPE_BUFFER, &filterDmlTensorDesc->bufferDesc};
+            }
+
+            std::vector<std::shared_ptr<EdgeInfoBase>> inputEdges = {inputEdge, filterEdge};
+
+            const DML_TENSOR_DESC* biasTensorDescPtr = nullptr;
+            DML_TENSOR_DESC newBiasTensorDesc = {};
+            if (options->bias != nullptr) {
+                DAWN_ASSERT(mGraphEdgesMap.find(inputsOperand[2].Get()) != mGraphEdgesMap.end());
+                auto biasEdge = mGraphEdgesMap[inputsOperand[2].Get()];
+                auto biasDims = ConvertDimensions(convTranspose2d->Inputs()[2].Get()->Shape());
+                if (biasDims[0] != newFilterDims[0] || biasDims.size() != 1) {
+                    return DAWN_INTERNAL_ERROR(
+                        "The bias should be 1-D tensor with the shape of [output_channels].");
+                }
+                // Reshape bias from 1-D to 4-D for NCHW layout.
+                std::vector<UINT> newBiasDims = {1, biasDims[0], 1, 1};
+                std::shared_ptr<DmlTensorDesc> biasDmlTensorDesc(new DmlTensorDesc);
+                if (!CreateDmlTensorDesc(mDmlTensorsDesc, biasDmlTensorDesc,
+                                         &biasEdge->outputTensorDESC, newBiasDims)) {
+                    return DAWN_INTERNAL_ERROR("Failed to create DML tensor description.");
+                }
+                newBiasTensorDesc = {DML_TENSOR_TYPE_BUFFER, &biasDmlTensorDesc->bufferDesc};
+                biasTensorDescPtr = &newBiasTensorDesc;
+                inputEdges.push_back(biasEdge);
+            }
+
+            std::vector<UINT> outputDims(4);
+            if (options->outputSizes != nullptr) {
+                std::vector<UINT> outputSizes;
+                outputSizes.assign(options->outputSizes,
+                                   options->outputSizes + options->outputSizesCount);
+                if (options->inputLayout == wnn::InputOperandLayout::Nchw) {
```

Comment:
Could we get shape here like line 1856? You know we have calculated the output shape of each op.

Reply:
Unlike conv2d, the outputSizes here is a parameter, and we can only get it directly if it is not set.